pkg/controller/statushandler/handler.go

```diff
@@ -18,6 +19,10 @@ func Update(ctx *workflow.Context, kubeClient client.Client, eventRecorder recor
 	resource.UpdateStatus(ctx.Conditions(), ctx.StatusOptions()...)
 
 	if err := patchUpdateStatus(kubeClient, resource); err != nil {
+		// if the resource has been deleted, it will not be possible to update the status.
+		if apiErrors.IsNotFound(err) {
```

Comment:
Great fix 👍 
What do you think about adding a special log message?
The comment which you've added is close to the perfect one for it, we can swap the comment for a log message.

Reply:
We can add a log line indicating this has happened, that's a good idea 👍